### PR TITLE
Fix issue #526: wrong quote position writing condensed flow

### PR DESCRIPTION
**Pull request by GabrieleC**

### CODE REVIEW

File: lib/js-yaml/dumper.js [hunk 1]
```diff
@@ -569,9 +569,8 @@ function writeFlowMapping(state, level, object) {
       pairBuffer;
 
   for (index = 0, length = objectKeyList.length; index < length; index += 1) {
-    pairBuffer = state.condenseFlow ? '"' : '';
-
-    if (index !== 0) pairBuffer += ', ';
+    pairBuffer = index === 0 ? '' : ', ';
```
Comment (puzrin) [inline lib/js-yaml/dumper.js]: Please avoid such notation, it's not enough easy to read. Use `if (index !== 0) pairBuffer += ', ';` and so on in appropriate places.

**Comment by puzrin** [comment]
Seems ok in general. Please take a look at comment & squash all to single commit.

**Comment by GabrieleC** [comment]
> 
> 
> Seems ok in general. Please take a look at comment & squash all to single commit.

@puzrin Improved readability and squashed commits.

**Comment by puzrin** [comment]
Thank you very much!

**Pull request merged**